### DISCUSSION TITLE
Create English version of gme file easily?

### DISCUSSION THREAD

**Discussion by Germling** [entropia/tip-toi-reveng#297] [Ideas]
Hello, my idea is to build a Python script to identify all OGG files that have speech, convert them to text, then translate that text and generate audio files again. This way, I could easily translate common books like "Auf dem Bauernhof" into English. I just have to overwrite the OGG files and build a new GME.

Main challenges: 
- Identify only those OGG files with speech
- Batch process these files using various TTS and translation APIs

**Comment by uli42** [comment]
I like this idea, and it could be extended to add new languages to the
tiptoi universe. One problem might be to identify different speakers to
generate different voices.

Interestingly the gme files support English as a language but there seem to
be no products on the market that actually use it.

**Reply by Germling** [reply]
[ogg_batch_translation_silero.zip](https://github.com/user-attachments/files/17291251/ogg_batch_translation_silero.zip)
I wrote a Python script that can run through the whole media folder, scan for speech via Silero VAD, convert German speech to text, translate the German text to English and convert the English text to speech. An example is attached.
The output file is temp.mp3; as you can hear, it uses a female voice by default, and there are nicer-sound voices available.

I think the speaker types could be identified via VAD and mapped in some way. But the challenge are large ogg files with lots of background noise. I don't know how this could be preserved... You would need to isolate only the speech somehow.

**Comment by uli42** [comment]
Maybe this could be solved using something like
https://github.com/noisetorch/NoiseTorch

**Comment by sum12** [comment]
I did build this, using some bunch of bash script and speaches.ai (identify if there is any german in there) and chatterbox-TTS (for VoiceCloning, was couple of hours using a 4090). Got pretty good results. But I noticed the `tttool media` does not dump all audio. 

Pretty certain because, the "Mein Wörter-Bilderbuch XXL 49257" seems to have additional audio like "can you find all the elephants" and none of the ~1330 media files referred to them. (unless I missed it) 

There are 7/8 games bins in there and this additional media seems to be used only via games. 
I wonder if deciphering those media location helps uncover hints for games.
